### PR TITLE
Yamaha zones: Add representation-property

### LINKED ISSUE #4259: yamahareceiver - Disable Auto Discovery (or making it smarter?)
Hi,

I am configuring some pre-defined things and Items for my OpenHAB and I would like to disable auto discovery when a spcific "thing" is already configured (in this particular case, a Yamaha Amplifier).

What I did is to create a services/yamaha.cfg with this content:
org.openhab.yamahareceiver:enableAutoDiscovery=false

Following a suggestion for a different binding, but it doesn't seems to work, therefore my question, is that possible? Can be implemented?

Also, maybe stupid maybe not, does exist a config parameter that to "don't show already configured things"? (example matching IDs).

Thanks 
Best regards
Andrea

### PR DESCRIPTION
**Pull request by davidgraeff**
Fix #4259

Signed-off-by: David Graeff <david.graeff@web.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public static ThingUID zoneThing(ThingUID bridgeUid, String zoneName) {
      * The available zones are within the {@link DeviceInformationState}. Will will publish those
      * as things via this discovery service instance.
      *
-     * @param state The device information state
+     * @param state     The device information state
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java]: Something went wrong with your formatter.

Comment (davidgraeff) [reply addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java]: New installed eclipses never apply the ESH formatter settings correctly :/

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java]: Yes, I have heard about that: https://github.com/eclipse/smarthome/issues/6424

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java [hunk 2]
```diff
@@ -90,12 +90,10 @@ public void publishZones(DeviceInformationState state, ThingUID bridgeUid) {
...
 
-            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(uid)
-                    .withProperties(properties)
-                    .withLabel(state.name + " " + zoneName)
-                    .withBridge(bridgeUid)
-                    .build();
-            
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withLabel(state.name + " " + zoneName).withBridge(bridgeUid)
+                    .withRepresentationProperty(CONFIG_ZONE).build();
```
Comment (wborn) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java]: Can you also add the representation property to the `zone` thing type definition? See also the [ESH documentation](https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#representation-property).

**Comment by davidgraeff** [comment]
Guys can we apply this PR without the representation-property-tag in the xml and add that in another PR?
Actually all my bindings even the ESH ones have that tag *not* defined yet. I have this branch not at hand right now and cannot add it in time for OH 2.4.

**Comment by wborn** [comment]
I fixed it @davidgraeff! :wink:

**Pull request merged**